src/frontend/http.rs

```diff
@@ -100,6 +102,23 @@ fn plan_to_etag(plan: &LogicalPlan) -> String {
     encode(hasher.finalize())
 }
 
+// Construct a content-type header value that also includes schema information.
+fn content_type_with_schema(schema: SchemaRef) -> HeaderValue {
+    let schema_string = schema_to_json(schema.as_ref()).to_string();
+    let output = utf8_percent_encode(&schema_string, NON_ALPHANUMERIC);
+
+    match HeaderValue::from_str(
+        format!("application/json; arrow-schema-escaped={output}").as_str(),
+    ) {
+        Ok(value) => value,
+        Err(err) => {
+            // Seems silly to error out here if the query itself succeeded.
+            warn!("Couldn't generate content type header for output schema {output}: {err:?}");
+            HeaderValue::from_static("application/json")
+        }
+    }
```

Comment:
minor: [`unwrap_or_else`](https://doc.rust-lang.org/std/result/enum.Result.html#method.unwrap_or_else) could work here (no idea if you tried it; I haven't tested this code)

```suggestion
    HeaderValue::from_str(
        format!("application/json; arrow-schema-escaped={output}").as_str(),
    ).unwrap_or_else(|e| {
        // Seems silly to error out here if the query itself succeeded.
        warn!("Couldn't generate content type header for output schema {output}: {e:?}");
        HeaderValue::from_static("application/json")
    })
```